src/interpreter/blockchain.rs

```diff
@@ -100,9 +102,7 @@ where
 
         // update frame pointer, if we have a stack frame (e.g. fp > 0)
         if fp > 0 {
-            let fpx = fp
-                .checked_add(CallFrame::code_size_offset())
-                .ok_or_else(|| Bug::new(BugId::ID008, BugVariant::FramePointerOverflow))?;
+            let fpx = checked_add_usize(fp, CallFrame::code_size_offset())?;
```

Comment:
This can't overflow unless we have a bug. Consider using `add_usize` instead. If the `$fp` will always point to a valid memory address, and we can't overflow `usize` or `u64` with a valid memory address + frame offset.

Consider using `add_usize` instead.
https://github.com/FuelLabs/fuel-vm/blob/2053b2fd2f2e874670a11cac4158c7ec88397576/src/arith.rs#L73

 otherwise, might be worthy to add a new function to `arith.rs` that will fail only if we have a bug, using the same signature as `add_word`

https://github.com/FuelLabs/fuel-vm/blob/2053b2fd2f2e874670a11cac4158c7ec88397576/src/arith.rs#L39

But then this function will contain a hint that, whenever it is reached, it will necessarily represent a bug in the code - so we output an according error message

Reply:
Could be smth like `add_usize_infallible(usize, usize) -> Result<usize, RuntimeError>`

Reply:
I agree and have modified this to use add_usize